Remove dynamic manually allocated memory

payetvin opened the pull request.

payetvin committed:
sets byIndex

payetvin committed:
sets namebyIndex

payetvin committed:
correlation montlhy

payetvin committed:
correlation annual

payetvin committed:
small refactor in correlation

payetvin committed:
atsp tmparray

payetvin committed:
scenario rules

payetvin committed:
xcast in ts generator

payetvin committed:
Revert "xcast in ts generator"

This reverts commit 065acfbb7e5da16190174abeaace4d87977acf15.

payetvin committed:
use make_shared for sets

payetvin committed:
Merge branch 'develop' into feature/remove-new

payetvin committed:
link info in cbuilder

payetvin committed:
thermal cluster prepro

payetvin committed:
format

JasonMarechal25 committed:
Use unique_ptr

JasonMarechal25 committed:
Remove this

flomnes reviewed (commented):
`Correlation::retrieveMontlyMatrixArray` is unused, consider removing it. It would be a good thing since `std::unique_ptr::get` is "dangerous", it can easily lead to dangling pointers
```cpp
auto ptr = std::make_unique<Cow>();
Cow* cow = ptr.get();
// ptr goes out of scope, is deleted
cow->moo(); // segfault
```

JasonMarechal25 committed:
Use unique_ptr

JasonMarechal25 committed:
Use uniqie_ptr

JasonMarechal25 commented on src/libs/antares/study/include/antares/study/study.h:
```diff
@@ -586,7 +580,7 @@ class Study: public Yuni::NonCopyable<Study>, public LayerData
     //@{
     //! Rules for building scenarios (can be null)
-    ScenarioBuilder::Sets* scenarioRules = nullptr;
+    std::shared_ptr<ScenarioBuilder::Sets> scenarioRules;
```
Can't be declared by value easily. Create circular include dependency. Would require classes in antares/study/scenario-builder to be split header/impl

JasonMarechal25 committed:
Remove dynamic allocations (#2275)

JasonMarechal25 committed:
Fix build

JasonMarechal25 committed:
Fix potential access violation (#2278)

We create a shared_ptr on the stack, stock the raw pointer in a vector
then exit the scop. At this point the pointer is not valid anymore

payetvin committed:
Merge branch 'develop' into feature/remove-new
